cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -54,6 +54,28 @@ def _cleanup():
     return vm
 
 
+@pytest.fixture(scope="function")
+def test_vm_tools(request, provider, vm_name, full_template):
+    """Fixture to provision vm with preinstalled tools to the provider being tested"""
+    vm = VM.factory(vm_name, provider, template_name=full_template["name"])
+    logger.info("provider_key: %s", provider.key)
+
+    @request.addfinalizer
+    def _cleanup():
+        vm.delete_from_provider()
+        if_scvmm_refresh_provider(provider)
+
+    if not provider.mgmt.does_vm_exist(vm.name):
+        logger.info("deploying %s on provider %s", vm.name, provider.key)
+        vm.create_on_provider(allow_skip="default",
+                              find_in_cfme=True)
+    else:
+        logger.info("recycling deployed vm %s on provider %s", vm.name, provider.key)
```

Comment:
I think reusing would be better here, recycling gives me the feeling it is somehow reset